src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs

```diff
@@ -48,29 +48,38 @@ public async Task<List<PSCommandMessage>> Handle(GetCommandParams request, Cance
             PSCommand psCommand = new PSCommand();
 
             // Executes the following:
-            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Select-Object -Property Name,ModuleName | Sort-Object -Property Name
+            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Sort-Object -Property Name
             psCommand
                 .AddCommand("Microsoft.PowerShell.Core\\Get-Command")
                     .AddParameter("CommandType", new[] { "Function", "Cmdlet", "ExternalScript" })
-                .AddCommand("Microsoft.PowerShell.Utility\\Select-Object")
-                    .AddParameter("Property", new[] { "Name", "ModuleName" })
                 .AddCommand("Microsoft.PowerShell.Utility\\Sort-Object")
                     .AddParameter("Property", "Name");
 
-            IEnumerable<PSObject> result = await _powerShellContextService.ExecuteCommandAsync<PSObject>(psCommand).ConfigureAwait(false);
+            IEnumerable<CommandInfo> result = await _powerShellContextService.ExecuteCommandAsync<CommandInfo>(psCommand).ConfigureAwait(false);
 
             var commandList = new List<PSCommandMessage>();
             if (result != null)
             {
-                foreach (dynamic command in result)
+                foreach (CommandInfo command in result)
                 {
+                    // Get the default ParameterSet
+                    string defaultParameterSet = null;
+                    foreach (CommandParameterSetInfo parameterSetInfo in command.ParameterSets)
+                    {
+                        if (parameterSetInfo.IsDefault)
+                        {
+                            defaultParameterSet = parameterSetInfo.Name;
+                            break;
+                        }
+                    }
+
                     commandList.Add(new PSCommandMessage
                     {
                         Name = command.Name,
                         ModuleName = command.ModuleName,
                         Parameters = command.Parameters,
                         ParameterSets = command.ParameterSets,
-                        DefaultParameterSet = command.DefaultParameterSet
```

Comment:
What's the story on needing the for-loop instead of this?

Reply:
it's a NoteProperty and doesn't actually exist on the type.

Reply:
CommandInfo's only have:

https://github.com/PowerShell/PowerShell/blob/b1e998046e12ebe5da9dee479f20d479aa2256d7/src/System.Management.Automation/engine/CommandInfo.cs#L611

no `DefaultParameterSet`

Reply:
`DefaultParameterSet`'s _might_ only exist on `CmdletInfo` and `FunctionInfo`.. so I could special case those types... but there are a lot of `*Info`'s

Reply:
Given that we're probably mostly going to deal with those types, I think that would make sense as an optimisation

Reply:
added a switch but still kept the loop for everything else (which atm is just ExternalScript).